Source/com/drew/metadata/eps/EpsReader.java

```diff
@@ -0,0 +1,466 @@
+    {
+        try {
+            String[] tag = new String[]{"", ""};
+            tag[0] += (char)reader.getByte(pos);
+            pos++;
+
+            // Loop until data/binary starts OR available data has run out (not accurate, but should not be an issue as we are only interested in the beginning
+            while (pos < reader.getLength() && tag[0].hashCode() != EpsDirectory.TAG_BEGIN_BINARY
+                && tag[0].hashCode() != EpsDirectory.TAG_BEGIN_DATA
+                && tag[0].hashCode() != EpsDirectory.TAG_AI9_END_PRIVATE_DATA)
+            {
+                // Check if the comment has been found in the tag map - if so, extract the data
+                if (EpsDirectory._tagNameMap.containsKey(tag[0].hashCode())) {
+                    if (tag[0].hashCode() == EpsDirectory.TAG_DSC_VERSION) {
+                        extractDscData(directory, tag);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_IMAGE_DATA) {
+                        extractImageData(directory);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_BEGIN_PHOTOSHOP) {
+                        extractPhotoshopData(metadata);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_BEGIN_ICC) {
+                        extractIccData(metadata);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_BEGIN_XML_PACKET) {
+                        extractXmpData(metadata);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_CONTINUE_LINE) {
+                        extractContinuation(directory, previousTag);
+                    } else {
+                        extractData(directory, tag);
+                    }
+                    if (tag[0].hashCode() != EpsDirectory.TAG_CONTINUE_LINE)
+                        previousTag = tag[0].hashCode();
+                    tag = new String[]{"", ""};
+                    tag[0] = "" + (char) reader.getByte(pos);
+                    pos++;
+                } else {
+                    previousTag = Integer.MAX_VALUE;
+                }
+
+                // Comments should never contain spaces/returns as per the EPS conventions
+                switch (tag[0].charAt(tag[0].length() - 1)) {
+                    case 0xA:
+                    case 0xD:
+                    case 0x9:
+                    case 0x20:
+                        tag[0] = "" + (char) reader.getByte(pos);
+                        pos++;
+                        while (tag[0].charAt(0) != 0x25) {
+                            tag[0] = "" + (char) reader.getByte(pos);
+                            pos++;
+                        }
+                        break;
+                    default:
+                        tag[0] += (char) reader.getByte(pos);
+                        pos++;
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
The only (quick) alternative to `getHashCode` that comes to mind is to use arbitrary (unique, incrementing) integers, and make another map from the EPS string to that int. When reading metadata, the int would be looked up, and used for storage. The existing map from int to friendly string would remain.

This wouldn't work for unknown tags though, as you wouldn't be able to deduce an int, and they'd have to be omitted (unlike TIFF which does support unknown tags as there's always an int available). The library started off focusing on Exif, and the use of int keys harks back to those days. I look forward to improving the design some day as it's too limiting as is.